openupgrade_scripts/scripts/account/15.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,429 @@
+        env.cr,
+        """
+        UPDATE account_move am
+        SET always_tax_exigible =
+            CASE
+                WHEN (SELECT COUNT(aml.currency_id)
+                     FROM account_move_line aml
+                     WHERE aml.move_id = am.id) > 1
+                THEN true
+                ELSE false
+            END
+        WHERE am.always_tax_exigible IS NULL""",
+    )
+    # 2. Set always_tax_exigible = False
+    #    if record._collect_tax_cash_basis_values() is True
+    # 2.2 Set always_tax_exigible = False
+    #     if any(line.account_internal_type in ('receivable', 'payable')
+    #     else True
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move am
+        SET always_tax_exigible =
+            CASE
+                WHEN (
+                    SELECT COUNT(*)
+                    FROM account_move_line aml
+                    JOIN account_account aa ON aa.id = aml.account_id
+                    JOIN account_account_type aat ON aat.id = aa.user_type_id
+                    WHERE aml.move_id = am.id
+                    AND aat.type IN ('receivable', 'payable')) > 0
+                THEN false
+                ELSE true
+            END
+        WHERE am.always_tax_exigible IS NULL""",
+    )
+    # 2. Set always_tax_exigible = False
+    #    if record._collect_tax_cash_basis_values() is True
+    # 2.3 Set always_tax_exigible = False
+    #     if any(line.tax_line_id.tax_exigibility == 'on_payment')
+    #     else True
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move am
+        SET always_tax_exigible =
+            CASE
+                WHEN (
+                    SELECT COUNT(*)
+                    FROM account_move_line aml
+                    JOIN account_tax tax ON tax.tax_exigibility = 'on_payment'
+                        AND aml.tax_line_id = tax.id
+                    WHERE aml.move_id = am.id) > 0
+                THEN false
+                ELSE true
+            END
+        WHERE am.always_tax_exigible IS NULL""",
+    )
+    # 2. Set always_tax_exigible = False
+    #    if record._collect_tax_cash_basis_values() is True
+    # 2.4 Set always_tax_exigible = False
+    #     if any(line.tax_ids.tax_exigibility == 'on_payment')
+    #     else True
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move am
+        SET always_tax_exigible =
+            CASE WHEN (
+                SELECT COUNT(aml_tax_rel.account_tax_id)
+                FROM account_move_line_account_tax_rel aml_tax_rel
+                JOIN account_move_line aml ON aml.id = aml_tax_rel.account_move_line_id
+                    AND aml.move_id = am.id
+                JOIN account_tax tax ON tax.tax_exigibility = 'on_payment'
+                    AND tax.id = aml_tax_rel.account_tax_id) > 0
+            THEN false
+            ELSE true
+            END
+        WHERE am.always_tax_exigible IS NULL""",
+    )
```

Comment:
This query does not handle the case when the child taxes of taxes on the line have `always_tax_exigible` is `on_payment`
https://github.com/odoo/odoo/blob/15.0/addons/account/models/account_move.py#L2470:L2472